src/main/scala/ai/eto/rikai/sql/spark/datasources/VideoDataSourceV2.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Rikai authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.eto.rikai.sql.spark.datasources
+
+import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.execution.datasources.FileFormat
+import org.apache.spark.sql.execution.datasources.v2.FileDataSourceV2
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+class VideoDataSourceV2 extends FileDataSourceV2 {
+  override def shortName(): String = "video"
```

Comment:
This means EACH video is ONE dataframe? Is this the desired use case?

cc @changhiskhan ?

Reply:
**Using the video format data source**, each video will be loaded as dataframe.

It is one of the use cases.